core/slice.go

```diff
@@ -1169,7 +1169,10 @@ func (sl *Slice) loadLastState() error {
 	if sl.ProcessingState() {
 		sl.miner.worker.LoadPendingBlockBody()
 	}
-	sl.WriteBestPh(rawdb.ReadBestPendingHeader(sl.sliceDb))
+	bestPh := rawdb.ReadBestPendingHeader(sl.sliceDb)
+	if bestPh != nil {
+		sl.WriteBestPh(bestPh)
```

Comment:
Seems like it would be better to include this nil check in sl.WriteBestPh itself, as its called in other places and we may forget to include the nil check.

Reply:
added